src/prefect/deployments/deployments.py

```diff
@@ -593,97 +598,28 @@ def _validate_schedule(cls, value):
     )
 
     @validator("infrastructure", pre=True)
-    def infrastructure_must_have_capabilities(cls, value):
-        if isinstance(value, dict):
-            if "_block_type_slug" in value:
-                # Replace private attribute with public for dispatch
-                value["block_type_slug"] = value.pop("_block_type_slug")
-            block = Block(**value)
-        elif value is None:
-            return value
-        else:
-            block = value
-
-        if "run-infrastructure" not in block.get_block_capabilities():
-            raise ValueError(
-                "Infrastructure block must have 'run-infrastructure' capabilities."
-            )
-        return block
+    def validate_infrastructure_capabilities(cls, value):
+        return infrastructure_must_have_capabilities(value)
 
     @validator("storage", pre=True)
-    def storage_must_have_capabilities(cls, value):
-        if isinstance(value, dict):
-            block_type = Block.get_block_class_from_key(value.pop("_block_type_slug"))
-            block = block_type(**value)
-        elif value is None:
-            return value
-        else:
-            block = value
-
-        capabilities = block.get_block_capabilities()
-        if "get-directory" not in capabilities:
-            raise ValueError(
-                "Remote Storage block must have 'get-directory' capabilities."
-            )
-        return block
+    def validate_storage(cls, value):
+        return storage_must_have_capabilities(value)
 
     @validator("parameter_openapi_schema", pre=True)
-    def handle_openapi_schema(cls, value):
-        """
-        This method ensures setting a value of `None` is handled gracefully.
-        """
-        if value is None:
-            return ParameterSchema()
-        return value
+    def validate_parameter_openapi_schema(cls, value):
+        return handle_openapi_schema(value)
```

Comment:
Current state with this PR:
```python
parameter_openapi_schema: ParameterSchema = Field(
        default_factory=ParameterSchema,
        description="The parameter schema of the flow, including defaults.",
    )

@validator("parameter_openapi_schema", pre=True)
def validate_parameter_openapi_schema(cls, value):
        return handle_openapi_schema(value)
```

Once we drop support for v1, we can do something like 

validators.py
```python
ParameterOpenAPISchema = Annotated[
    ParameterSchema,
    Field(default_factory=ParameterSchema, description="The parameter schema of the flow, including defaults."),
    BeforeValidator(handle_openapi_schema)
]
```

deployments.py
```python
parameter_openapi_schema: ParameterOpenAPISchema
```

Reply:
oh the future is gonna be pretty